src/Form/GroupSubscribeForm.php

```diff
@@ -23,23 +26,50 @@ class GroupSubscribeForm extends ContentEntityForm {
   /**
    * OG access service.
    *
-   * @var \Drupal\og\OgAccess
+   * @var \Drupal\og\OgAccessInterface
    */
   protected $ogAccess;
 
   /**
-   * Constructs a SubscriptionController object.
+   * Constructs a GroupSubscribeForm.
+   *
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The OG access service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface|\Drupal\Core\Entity\EntityManagerInterface $entity_repository
+   *   The entity repository service.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   *
+   * @todo Set the correct type hint on the second argument when Drupal 8.6.0 is
```

Comment:
if we'll do that, then it means OG will support only 8.6.0 - or you mean that 8.6.0 will be responsible for the BC?

Reply:
Yes, we have to leave out the type hint now to make it work for both 8.5 and 8.6. Once 8.5 is deprecated we don't need this B/C workaround any more and can add the `EntityRepositoryInterface` type hint. But in doing so it indeed means that OG will only support Drupal 8.6 from that point onwards, so this should be reflected in `composer.json`.

To clarify this I will create the followup issue to handle this task and link it here in the `@todo`.